lib/jekyll/document.rb

```diff
@@ -256,7 +256,7 @@ def published?
     # values
     #
     # Returns nothing.
-    def read(opts = {})
+    def read(opts = {}, path: self.path)
```

Comment:
Hacky code 1: I needed to add this optional parameter since `Page.path` is actually the relative path, while `read` needs the full path. So when `read` is called in the Page constructor, I pass in the full path.